### PR TITLE
CFE-2889: cf-remote package selection, --bootstrap, --demo etc.

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a PR! Maybe @vpodzime can review this?

File: contrib/cf-remote/cf_remote/packages.py [hunk 1]
```diff
@@ -117,6 +117,13 @@ def add_tags_from_filename(self, filename):
         else:
             self.add_tag("client")
             self.add_tag("agent")
+        parts = filename.split(".")
+        if "x86_64" in parts:
+            self.add_tag("x86_64")
+            self.add_tag("64")
+        if "amd64" in parts:
+            self.add_tag("amd64")
+            self.add_tag("64")
```
Comment (vpodzime) [inline contrib/cf-remote/cf_remote/packages.py]: Should this be part of the previous commit?

File: contrib/cf-remote/cf_remote/remote.py [hunk 2]
```diff
@@ -148,15 +150,33 @@ def boootstrap_host(host, policy_server):
...
+        extension = None
+        if "dpkg" in data["bin"]:
+            extension = ".deb"
+        elif "rpm" in data["bin"]:
+            extension = ".rpm"
+        releases = Releases()
+        release = releases.default
+        artifacts = release.find(tags, extension)
+        if not artifacts:
+            user_error("Could not find an appropriate package for host")
```
Comment (vpodzime) [inline contrib/cf-remote/cf_remote/remote.py]: This should suggest using the right option to choose it manually.

File: contrib/cf-remote/cf_remote/remote.py [hunk 3]
```diff
@@ -148,15 +150,33 @@ def boootstrap_host(host, policy_server):
...
+        artifact = artifacts[-1]
```
Comment (vpodzime) [inline contrib/cf-remote/cf_remote/remote.py]: Any reason why it's taking the last one?

File: contrib/cf-remote/cf_remote/demo.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        command = "/var/cfengine/bin/cf-agent -K"
+        output = remote.ssh_sudo(connection, command)
+
+
+def disable_password_dialog(host):
+    print("Disabling password change on hub: '{}'".format(host))
+    api = "https://{}/api/user/admin".format(host)
+    d = json.dumps({"password": "password"})
+    creds = "admin:admin"
+    c = "curl -X POST  -k {} -k -u {}  -H 'Content-Type: application/json' -d '{}'".format(
```
Comment (vpodzime) [inline contrib/cf-remote/cf_remote/demo.py]: Double `-k`.

File: contrib/cf-remote/cf_remote/demo.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            "control_executor_splaytime": "1"
+        }
+    }
+
+
+def install_def_json(host):
+    print("Transferring def.json to hub: '{}'".format(host))
+    data = json.dumps(def_json(), indent=2)
+    path = os.path.join(cf_remote_dir("json"), "def.json")
+    save_file(path, data)
```
Comment (vpodzime) [inline contrib/cf-remote/cf_remote/demo.py]: Is this going to write the file locally? Where? Does it need *root* permissions to do so?

File: contrib/cf-remote/cf_remote/remote.py [hunk 6]
```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env python3
 import sys
+import os
 from os.path import basename
 from collections import OrderedDict
+import json
```
Comment (vpodzime) [inline contrib/cf-remote/cf_remote/remote.py]: I don't see neither `os` nor `json` in the added lines below. I guess some remnants of placing the code into a different file eventually?

File: contrib/cf-remote/cf_remote/demo.py (hunk 5 above)
Comment (olehermanse) [reply contrib/cf-remote/cf_remote/demo.py]: cf_remote_dir is in home folder `~/.cfengine/cf-remote/`. This writes `~/.cfengine/cf-remote/json/def.json`, similar to `~/.cfengine/cf-remote/json/releases.json`.

**Comment by olehermanse** [comment]
This pull request **introduces 1 alert** when merging 5fed8028a83120f27ca4d1ed9c04437442f20e4e into b15e9be2c0f661264d3da0e7529bc7973c9db3db - [view on LGTM.com](https://lgtm.com/projects/g/cfengine/core/rev/pr-bdb70fd6dc8d34ebdb05bceddccd7094958632c5)

**new alerts:**

* 1 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: contrib/cf-remote/cf_remote/packages.py (hunk 1 above)
Comment (olehermanse) [reply contrib/cf-remote/cf_remote/packages.py]: Fixed.

**Comment by olehermanse** [comment]
@vpodzime I believe I have addressed all your comments.

File: contrib/cf-remote/cf_remote/demo.py (hunk 5 above)
Comment (vpodzime) [reply contrib/cf-remote/cf_remote/demo.py]: OK :+1:

**Pull request merged**